registry/txt_test.go

```diff
@@ -60,6 +60,26 @@ func testTXTRegistryNew(t *testing.T) {
 	_, err = NewTXTRegistry(p, "txt", "txt", "owner", time.Hour)
 	require.Error(t, err)
 
+	_, err = NewTXTRegistry(p, "@invalid#", "", "owner", time.Hour)
+	require.Error(t, err, `invalid TXT prefix provided, expected "[a-z0-9-.]+" got "@invalid#"`)
+	_, err = NewTXTRegistry(p, "", "@invalid#", "owner", time.Hour)
+	require.Error(t, err, `invalid TXT suffix provided, expected "[a-z0-9-.]+" got "@invalid#"`)
+
```

Comment:
If I add

```
	_, err = NewTXTRegistry(p, "valid....", "", "owner", time.Hour)
	require.Error(t, err)
```

That also looks invalid to me, but the test fails. Are you sure the check that we are doing is fine?

Reply:
Then we need to improve the regex that is used. Any suggestions?

Reply:
@ruudk any thoughts on the suggestion from @Raffo